odoo/addons/base/models/ir_ui_view.py

```diff
@@ -520,6 +520,12 @@ def write(self, vals):
         if 'arch_updated' not in vals and ('arch' in vals or 'arch_base' in vals) and 'install_filename' not in self._context:
             vals['arch_updated'] = True
 
+        if 'arch_db' in vals and self.type not in self._fields['type']._selection:
```

Comment:
This looks a lot like you are hiding some bug.  Why do you need this?

Reply:
Moreover, `self` may be a recordset with more than one record.  In that case, `self.type` will crash.